develop/nix/devshell.nix

```diff
@@ -0,0 +1,64 @@
+{ inputs
+, ...
+}:
+
+{
+  perSystem = { pkgs, lib, ... }: {
+    devshells.default =
+      let
+        rust-toolchain = with pkgs;
+          [
+            ((rust-bin.fromRustupToolchainFile ../../rust-toolchain).override {
+              extensions = [ "rust-src" "rust-analyzer" ];
+            })
+          ] ++ [
+            cargo-make
+            cargo-nextest
+            cargo-binstall
+            cargo-sort
+            typos
+          ];
```

Comment:
Is it possible to use `./risedev install-tools`? So that we can reduce the places we need to maintain the test/build/check dependencies.

Reply:
Since `cargo install` compiles from source, yes, it can be used by nix users too.

But I think we can still keep it here? These tools are no different from other dependencies, we don't need much maintenance, nix users should figure it out by themself.

Reply:
Since `cargo install` compiles from source, yes, nix users can use it. I'll delete these if you think it's better.